src/message_data/mod.rs

```diff
@@ -32,6 +32,18 @@ impl MessageData {
         }
     }
 
+    pub fn consume(&self, message_id: &str) -> Result<String, MessageDataError> {
+        let key = self.generate_key(message_id)?;
+        let key_str = key.as_str();
+        self.client
+            .get(key_str)
+            .map(|metadata| {
+                self.client.del::<&str, u8>(key_str).ok();
```

Comment:
My IDE freaks out over here: 
![r](https://user-images.githubusercontent.com/128755/41618463-3cb3aad6-73d1-11e8-85c2-00a350989969.jpg)

But seems like it's false to raise an error here?

Reply:
Hmmm. Is your IDE set up to use nightly Rust? And have you done a `rustup update` recently?

Reply:
(that's probably not it, I'm just thinking out loud tbh)

Reply:
It seems to be using latest Nightly in the IDE, but I will double check...

Reply:
It's fine in VSCode , *shrugs*